### PR TITLE
Fix: Path normalization for file exclusions

### PR DESCRIPTION
**Pull request by ketoprakmaster**
### Issue:
Passing file paths with leading ./ or mismatched slashes to --exclude-files caused exclusions to fail because of strict string matching against normalized paths.

### Changes:
- Standardized all file exclusion logic to use absolute, resolved paths via pathlib.
- Anchored relative exclusion patterns to the target directory (-d) rather than the current working directory.
- Refactored exclusion matching into a helper function file_matches_exclude to ensure the Directory Tree and File Aggregator behave identically.

### Added Tests:
Added regression tests covering:
-   `./`-prefixed paths
-   absolute paths
-   glob patterns
-   differing `cwd` vs `-d`
-   Windows-style separators (NOTE: windows only)

Most new tests fail on `main` and all pass with this patch applied.

with the help from both gemini and chatgpt

### CODE REVIEW

File: all_code.py [hunk 1]
```diff
@@ -231,6 +222,28 @@ def should_include_file(file_path):
     return rel_file_path in FILES_TO_INCLUDE
 
 
+def file_matches_exclude(filepath: str, exclude_file_globs, startpath: str) -> bool:
+    """
+    Match file against user-provided exclude patterns.
+    All comparisons are done on normalized absolute paths.
+    """
+    try:
```
Comment (hermesimi) [inline all_code.py:230]: This is a great issue you have found and I liked your solution.

One proposal here is to use only line 231, because after python 3.6, pathlib normalizes the path by default ([see](https://stackoverflow.com/questions/32358767/normalize-non-existing-path-using-pathlib-only)), but you can use the `strict=False` explicitly for the sake of documentation. I removed the try except and kept only the pathlib line and it passed all tests.

Also, maybe consider adding `.absolute()` too, it might be redundant but it catches some edge cases better depending on the platform.

```
abs_path = str(pathlib.Path(filepath).absolute().resolve(strict=False))
```

File: all_code.py [hunk 2]
```diff
@@ -408,8 +421,21 @@ def main():
...
+    # Normalize --exclude-files patterns (relative → startpath, remove ./, fix slashes)
+    exclude_file_globs = []
+    for raw in _split_csv(args.exclude_files):
+        pat = os.path.expanduser(raw)
+
+        # If relative, anchor to startpath (NOT cwd)
+        if not os.path.isabs(pat):
+            pat = os.path.join(startpath, pat)
+
+        try:
```
Comment (hermesimi) [inline all_code.py:433]: Same here.

**Review by hermesimi — APPROVED** [review]
Nice addition. Left some proposals but they are more formal than functional, so feel free to implement them or dismiss. 

Also, I have personally been using more and more pathlib than os.path, it would be a nice exercise for anyone in the future to refactor this to use pathlib since it's so much more reliable, but I digress.

LGTM

**Comment by ketoprakmaster** [comment]
Hello thanks for the response!

In regards to your proposal I do agree. As a new junior python developers I had started out with pathlib not os.path as a pathing library in python so I can't say much about the reliability comparing the two. But I do find it much simpler, modern and easier to follow so I do wonder why don't I see it everywhere.

I might consider refactor this further _(while at it let's also improve the unit test?)_ But this PR is also open to edit so feel free for other maintainer to add any necessary changes

**Review by foxalabs — APPROVED** [review]
Great work!
